test/sol6/multipleEpochRewardsClaimer.js

```diff
@@ -0,0 +1,227 @@
+                expiryTimestamp
+            );
+    
+            stakerPercentageInPrecision = precisionUnits.mul(rewardInBPS).div(BPS);
+            await mockKyberDao.setStakerPercentageInPrecision(stakerPercentageInPrecision);
+    
+            // setup feeHandler
+            feeHandler = await FeeHandler.new(daoSetter, proxy.address, kyberNetwork, knc.address, BURN_BLOCK_INTERVAL, daoOperator);
+            await feeHandler.setDaoContract(mockKyberDao.address, {from: daoSetter})
+            await feeHandler.setBurnConfigParams(sanityRate.address, ethWeiToBurn, {from: daoOperator});
+            await feeHandler.getBRR();
+    
+            // setup rewardsClaimer
+            rewardsClaimer = await MultipleEpochRewardsClaimer.new(mockKyberDao.address);
+        });
+
+        beforeEach("send fees for numEpochs", async() => {
+            initialEpoch = (await mockKyberDao.epoch()).toNumber();
+            await sendFeesToFeeHandler(mockKyberDao, feeHandler, numEpochs);
+            currentEpoch = (await mockKyberDao.epoch()).toNumber();
+        });
+
+        it("should not do anything if empty array is passed", async() => {
+            let stakerBalance = await Helper.getBalancePromise(staker);
+            await rewardsClaimer.claimMultipleRewards(feeHandler.address, [], {from: staker});
+            Helper.assertLesser(await Helper.getBalancePromise(staker), stakerBalance, "staker balance changed");
+        });
+
+        it("should claim for 1 epoch", async() => {
+            let stakerBalance = await Helper.getBalancePromise(staker);
+            await rewardsClaimer.claimMultipleRewards(feeHandler.address, [initialEpoch], {from: staker});
+            Helper.assertGreater(await Helper.getBalancePromise(staker), stakerBalance, "staker balance did not increase");
+            assert.isTrue(await feeHandler.hasClaimedReward(staker, initialEpoch));
+        });
+
+        it("should claim for multiple epochs", async() => {
+            let stakerBalance = await Helper.getBalancePromise(staker);
+            await rewardsClaimer.claimMultipleRewards(feeHandler.address, [initialEpoch, currentEpoch - 2, currentEpoch - 1], {from: staker});
+            Helper.assertGreater(await Helper.getBalancePromise(staker), stakerBalance, "staker balance did not increase");
+            assert.isTrue(await feeHandler.hasClaimedReward(staker, initialEpoch));
+            assert.isTrue(await feeHandler.hasClaimedReward(staker, currentEpoch - 2));
+            assert.isTrue(await feeHandler.hasClaimedReward(staker, currentEpoch - 1));
+        });
+
+        it("should revert for invalid feeHandler address", async() => {
+            await expectRevert.unspecified(
+                rewardsClaimer.claimMultipleRewards(staker, [initialEpoch], {from: staker})
+            );
+        });
+    });
+});
+
+async function sendFeesToFeeHandler(dao, feeHandler, numEpochs) {
+    for (let i = 0; i < numEpochs; i++) {
+        sendVal = oneEth.add(oneEth);
```

Comment:
I am missing something
how do u make sure the staker is eligible for rewards for each epoch?

Reply:
just set a reward percentage for a staker in MockKyberDao

Reply:
As explained below

Reply:
I know how its done.
didn't see where. see it now.